bin/migrate-oats-data/noi/noi_decisions/noi_components/notice_of_intent_decision_components_soil_details_update.py

```diff
@@ -0,0 +1,146 @@
+                    )
+                except Exception as err:
+                    logger.exception(err)
+                    conn.rollback()
+                    failed_updates = count_total - successful_updates_count
+                    last_component_id = last_component_id + 1
+
+    logger.info(
+        f"Finished {etl_name}: total amount of successful updates {successful_updates_count}, total failed updates {failed_updates}"
+    )
+
+
+_soil_fill_query = """
+                    UPDATE alcs.notice_of_intent_decision_component 
+                    SET soil_to_place_volume = %(volume)s
+                        , soil_to_place_area = %(project_area)s
+                        , soil_to_place_maximum_depth = %(depth)s
+                        , soil_to_place_average_depth = %(depth)s
+                        , soil_fill_type_to_place  = %(type)s
+                    WHERE alcs.notice_of_intent_decision_component.oats_alr_appl_component_id = %(oats_alr_appl_component_id)s
+"""
+
+_soil_remove_query = """
+                    UPDATE alcs.notice_of_intent_decision_component 
+                    SET soil_to_remove_volume = %(volume)s
+                        , soil_to_remove_area = %(project_area)s
+                        , soil_to_remove_maximum_depth = %(depth)s
+                        , soil_to_remove_average_depth = %(depth)s
+                        , soil_type_removed  = %(type)s
+                    WHERE alcs.notice_of_intent_decision_component.oats_alr_appl_component_id = %(oats_alr_appl_component_id)s
+"""
+
+
+def _update_soil_fields(conn, batch_size, cursor, rows):
+    parsed_data_list = _prepare_oats_alr_decision_component_data(rows)
+
+    if len(parsed_data_list[SoilChangeCode.REMOVE.value]) > 0:
+        execute_batch(
+            cursor,
+            _soil_remove_query,
+            parsed_data_list[SoilChangeCode.REMOVE.value],
+            page_size=batch_size,
+        )
+
+    if len(parsed_data_list[SoilChangeCode.ADD.value]) > 0:
+        execute_batch(
+            cursor,
+            _soil_fill_query,
+            parsed_data_list[SoilChangeCode.ADD.value],
+            page_size=batch_size,
+        )
+
+    conn.commit()
+
+
+def _prepare_oats_alr_decision_component_data(row_data_list):
+    data_list = {SoilChangeCode.REMOVE.value: [], SoilChangeCode.ADD.value: []}
+    for row in row_data_list:
+        data = map_soil_fields(row)
+
+        if data["soil_change_code"] == SoilChangeCode.ADD.value:
+            data_list[SoilChangeCode.ADD.value].append(data)
+        elif data["soil_change_code"] == SoilChangeCode.REMOVE.value:
+            data_list[SoilChangeCode.REMOVE.value].append(data)
+
+    return data_list
+
+
+def map_soil_fields(data):
+    soil_type = data.get("material_desc", "")
+    soil_origin = data.get("material_desc", "")
```

Comment:
Won't these two variables have the exact same value?

Reply:
good catch, that is a leftover from soil mapping of noi component